foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/Fabric8LocationMigration.java

```diff
@@ -0,0 +1,148 @@
+package com.boozallen.aissemble.upgrade.migration.v1_8_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractPomMigration;
+import com.boozallen.aissemble.upgrade.util.AissembleFileUtils;
+import org.apache.maven.model.BuildBase;
+import org.apache.maven.model.InputLocation;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.PluginManagement;
+import org.apache.maven.model.Profile;
+import org.technologybrewery.baton.util.pom.PomHelper;
+import org.technologybrewery.baton.util.pom.PomModifications;
+import org.technologybrewery.baton.util.pom.PomModifications.Deletion;
+import org.technologybrewery.baton.util.pom.PomModifications.Insertion;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.boozallen.aissemble.upgrade.migration.v1_8_0.OrphedomosToFabric8Migration.FABRIC8_ARTIFACT_ID;
+import static org.apache.commons.lang3.StringUtils.repeat;
+import static org.technologybrewery.baton.util.pom.LocationAwareMavenReader.END;
+import static org.technologybrewery.baton.util.pom.LocationAwareMavenReader.START;
+
+/**
+ * In the move from Orphedomos to Fabric8, we can no longer specify the Docker plugin as part of the build for a module
+ * that is not a `docker-build` packaging type.  (Orphedomos allowed this and simply skipped all goals if the packaging
+ * type was not `orphedomos`.)  This migration ensures any `pom` type modules only specify the Docker plugin within a
+ * pluginManagement section.
+ */
+public class Fabric8LocationMigration extends AbstractPomMigration {
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        Model model = PomHelper.getLocationAnnotatedModel(file);
+        String packaging = model.getPackaging();
+        if (POM.equals(packaging)) {
+            boolean profilesMatch = model.getProfiles()
+                    .stream()
```

Comment:
Will this be fine in the event there are no profiles in the pom? Will `.getProfiles()` return null?

Reply:
`getProfiles` never returns null:
```java
public java.util.List<Profile> getProfiles() {
        if ( this.profiles == null )
        {
            this.profiles = new java.util.ArrayList<Profile>();
        }

        return this.profiles;
    }
```